blueoil/cmd/predict.py

```diff
@@ -13,21 +13,160 @@
+        raw_images.append(raw_image)
+
+    return np.array(images), np.array(raw_images)
+
+
+def _all_image_files(directory):
+    all_image_files = []
+    for file_path in glob(os.path.join(directory, "*")):
+        if os.path.isfile(file_path) and imghdr.what(file_path) in {"jpeg", "png"}:
+            all_image_files.append(os.path.abspath(file_path))
+
+    return all_image_files
+
+
+def _run(input_dir, output_dir, config, restore_path, save_images):
+    ModelClass = config.NETWORK_CLASS
+    network_kwargs = dict((key.lower(), val) for key, val in config.NETWORK.items())
+
+    graph = tf.Graph()
+    with graph.as_default():
+        model = ModelClass(
+            classes=config.CLASSES,
+            is_debug=config.IS_DEBUG,
+            **network_kwargs
         )
 
-    run_predict(input, output, experiment_id, None, restore_path, save_images)
+        is_training = tf.constant(False, name="is_training")
+
+        images_placeholder, _ = model.placeholders()
+        output_op = model.inference(images_placeholder, is_training)
+
+        init_op = tf.global_variables_initializer()
+
+        saver = tf.compat.v1.train.Saver(max_to_keep=None)
+
+    session_config = tf.ConfigProto()
+    sess = tf.Session(graph=graph, config=session_config)
+    sess.run(init_op)
+    saver.restore(sess, restore_path)
+
+    all_image_files = _all_image_files(input_dir)
+
+    step_size = int(math.ceil(len(all_image_files) / config.BATCH_SIZE))
+
+    writer = OutputWriter(
+        task=config.TASK,
+        classes=config.CLASSES,
+        image_size=config.IMAGE_SIZE,
+        data_format=config.DATA_FORMAT
+    )
+
+    results = []
+    for step in range(step_size):
+        start_index = (step) * config.BATCH_SIZE
+        end_index = (step + 1) * config.BATCH_SIZE
+
+        image_files = all_image_files[start_index:end_index]
+
+        while len(image_files) != config.BATCH_SIZE:
+            # add dummy image.
+            image_files.append(DUMMY_FILENAME)
+
+        images, raw_images = _get_images(
+            image_files, config.DATASET.PRE_PROCESSOR, config.DATA_FORMAT)
+
+        feed_dict = {images_placeholder: images}
+        outputs = sess.run(output_op, feed_dict=feed_dict)
+
+        if config.POST_PROCESSOR:
+            outputs = config.POST_PROCESSOR(outputs=outputs)["outputs"]
+
+        results.append(outputs)
+
+        writer.write(
+            output_dir,
+            outputs,
+            raw_images,
+            image_files,
+            step,
+            save_material=save_images
+        )
+
+    return results
+
+
+def run(input_dir, output_dir, experiment_id, config_file, restore_path, save_images):
+    environment.init(experiment_id)
+    config = config_util.load_from_experiment()
+    if config_file:
+        config = config_util.merge(config, config_util.load(config_file))
+
+    if not os.path.isdir(input_dir):
+        raise Exception("Input directory {} does not exist.".format(input_dir))
```

Comment:
Avoid raising `Exception` and use more specific ones. In this case,  `FileNotFoundError` might be a good choice